sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java

```diff
@@ -762,6 +868,23 @@ public void processElement(ProcessContext c) {
     }
   }
 
+  private static class ExtractOnlyPane<T> extends DoFn<Iterable<WindowedValue<T>>, Iterable<T>> {
+    @Override
+    public void processElement(ProcessContext c) throws Exception {
+      List<T> outputs = new ArrayList<>();
+      if (!Iterables.isEmpty(c.element())) {
+        long paneIndex = c.element().iterator().next().getPane().getIndex();
+        for (WindowedValue<T> value : c.element()) {
+          assertThat("Expected all elements to be in the same pane, but got different indicies",
```

Comment:
Updated the assertion to a checkState; specified that pane is both first and last, with an error message signifying that the trigger is expected to produce output exactly once.